### PR TITLE
FISH-8824: upgrading cdi 4.1 related dependencies and weld

### PR DESCRIPTION
**Pull request by breakponchito**
Upgrading dependencies for CDI 4.1 including WELD
## Description

This is a feature to upgrade CDI to version 4.1 and related dependencies, including WELD
### Testing Performed

Manual testing building the payara server without any issue and starting server on local environment with success with a deployment of a web application:
image
this is the web application deployed on the server with CDI 4.1 upgrades
[WepApp.zip](https://github.com/user-attachments/files/15893755/WepApp.zip)

### Testing Environment

Windows 11, Azul JDK 21, maven 3.9.5

### CODE REVIEW

File: appserver/packager/glassfish-ejb/pom.xml [hunk 1]
```diff
@@ -103,6 +103,10 @@
             <artifactId>ejb-http-admin</artifactId>
             <version>${project.version}</version>
         </dependency>
+        <dependency>
+            <groupId>jakarta.enterprise</groupId>
+            <artifactId>jakarta.enterprise.cdi-el-api</artifactId>
+        </dependency>
```
Comment (Pandrex247) [inline appserver/packager/glassfish-ejb/pom.xml:109]: Not sure this should be in here.
Would appserver-core, glassfish-common, or glassfish-jcdi not be a better place?

File: api/payara-api/pom.xml [hunk 2]
```diff
@@ -161,5 +161,25 @@
...
+            <artifactId>jakarta.authorization-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.validation</groupId>
+            <artifactId>jakarta.validation-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.security.enterprise</groupId>
+            <artifactId>jakarta.security.enterprise-api</artifactId>
+        </dependency>
```
Comment (Pandrex247) [inline api/payara-api/pom.xml:183]: Surely these are covered by the Platform API dependency?
I'd like it if we had a comment highlighting these dependencies as (presumably) a temporary measure
```suggestion
        <!-- Temporary dependencies while EE11 in development, remove once EE11 fully released -->
        <dependency>
            <groupId>jakarta.inject</groupId>
            <artifactId>jakarta.inject-api</artifactId>
        </dependency>
        <dependency>
            <groupId>jakarta.enterprise</groupId>
            <artifactId>jakarta.enterprise.cdi-api</artifactId>
        </dependency>
        <dependency>
            <groupId>jakarta.authorization</groupId>
            <artifactId>jakarta.authorization-api</artifactId>
        </dependency>
        <dependency>
            <groupId>jakarta.validation</groupId>
            <artifactId>jakarta.validation-api</artifactId>
        </dependency>
        <dependency>
            <groupId>jakarta.security.enterprise</groupId>
            <artifactId>jakarta.security.enterprise-api</artifactId>
        </dependency>
        <!-- End of temporary dependencies -->
```

File: appserver/payara-appserver-modules/microprofile/metrics/pom.xml [hunk 3]
```diff
@@ -138,7 +138,6 @@
         <dependency>
             <groupId>jakarta.servlet</groupId>
             <artifactId>jakarta.servlet-api</artifactId>
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/microprofile/metrics/pom.xml:140]: Comment highlighting this is temporary.
```suggestion
            <artifactId>jakarta.servlet-api</artifactId>
            <!-- Temporary compile dependency while EE11 in development, should get moved back to test scope once EE11 is finished -->
            <!-- <scope>test</scope> -->
```

File: appserver/payara-appserver-modules/microprofile/rest-client-ssl/pom.xml [hunk 4]
```diff
@@ -61,6 +61,10 @@ holder.
             <groupId>jakarta.platform</groupId>
             <artifactId>jakarta.jakartaee-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>jakarta.ws.rs</groupId>
+            <artifactId>jakarta.ws.rs-api</artifactId>
+        </dependency>
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/microprofile/rest-client-ssl/pom.xml:67]: Comment flag
```suggestion
        <!-- Temporary dependency while EE11 is in development. Should get removed once EE 11 is finished and platform API is updated -->
        <dependency>
            <groupId>jakarta.ws.rs</groupId>
            <artifactId>jakarta.ws.rs-api</artifactId>
        </dependency>
```

File: appserver/payara-appserver-modules/notification-jms-core/pom.xml [hunk 5]
```diff
@@ -62,5 +62,9 @@
             <groupId>jakarta.platform</groupId>
             <artifactId>jakarta.jakartaee-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>jakarta.jms</groupId>
+            <artifactId>jakarta.jms-api</artifactId>
+        </dependency>
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/notification-jms-core/pom.xml:68]:
```suggestion
        <!-- Temporary dependency while EE11 is in development. Should get removed once EE 11 is finished and platform API is updated -->
        <dependency>
            <groupId>jakarta.jms</groupId>
            <artifactId>jakarta.jms-api</artifactId>
        </dependency>
```

File: appserver/payara-appserver-modules/payara-micro-cdi/pom.xml [hunk 6]
```diff
@@ -86,6 +86,10 @@
             <groupId>jakarta.platform</groupId>
             <artifactId>jakarta.jakartaee-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>jakarta.enterprise.concurrent</groupId>
+            <artifactId>jakarta.enterprise.concurrent-api</artifactId>
+        </dependency>
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/payara-micro-cdi/pom.xml:92]:
```suggestion
        <!-- Temporary dependency while EE11 is in development. Should get removed once EE 11 is finished and platform API is updated -->
        <dependency>
            <groupId>jakarta.enterprise.concurrent</groupId>
            <artifactId>jakarta.enterprise.concurrent-api</artifactId>
        </dependency>
```

File: appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-service/pom.xml [hunk 7]
```diff
@@ -66,6 +66,10 @@
             <groupId>jakarta.platform</groupId>
             <artifactId>jakarta.jakartaee-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>jakarta.authentication</groupId>
+            <artifactId>jakarta.authentication-api</artifactId>
+        </dependency>
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-service/pom.xml:72]:
```suggestion
        <!-- Temporary dependency while EE11 is in development. Should get removed once EE 11 is finished and platform API is updated -->
        <dependency>
            <groupId>jakarta.authentication</groupId>
            <artifactId>jakarta.authentication-api</artifactId>
        </dependency>
```

File: appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-war/pom.xml [hunk 8]
```diff
@@ -71,6 +71,18 @@
...
+            <artifactId>jakarta.enterprise.cdi-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.ws.rs</groupId>
+            <artifactId>jakarta.ws.rs-api</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.ejb</groupId>
+            <artifactId>jakarta.ejb-api</artifactId>
+        </dependency>
```
Comment (Pandrex247) [inline appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-war/pom.xml:85]:
```suggestion
        <!-- Temporary dependencies while EE11 is in development. Should get removed once EE 11 is finished and web profile API is updated -->
        <dependency>
            <groupId>jakarta.enterprise</groupId>
            <artifactId>jakarta.enterprise.cdi-api</artifactId>
        </dependency>
        <dependency>
            <groupId>jakarta.ws.rs</groupId>
            <artifactId>jakarta.ws.rs-api</artifactId>
        </dependency>
        <dependency>
            <groupId>jakarta.ejb</groupId>
            <artifactId>jakarta.ejb-api</artifactId>
        </dependency>
        <!-- End of temporary dependencies -->
```

**Review by Pandrex247 — COMMENTED** [review]
I find it interesting we can't use the 11.0.0-M2 Platform API dependency...
Do the Platform API milestone releases not align with the [release plan milestones](https://jakartaee.github.io/platform/jakartaee11/JakartaEE11ReleasePlan)?

File: appserver/packager/glassfish-ejb/pom.xml (hunk 1 above)
Comment (breakponchito) [reply appserver/packager/glassfish-ejb/pom.xml]: yep I moved to the appserver-core, and worked correctly, thanks

File: api/payara-api/pom.xml (hunk 2 above)
Comment (breakponchito) [reply api/payara-api/pom.xml:184]: already commented

File: appserver/payara-appserver-modules/notification-jms-core/pom.xml (hunk 5 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/notification-jms-core/pom.xml:69]: already commented

File: appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-service/pom.xml (hunk 7 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-service/pom.xml:73]: already commented

File: appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-war/pom.xml (hunk 8 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/rest-monitoring/rest-monitoring-war/pom.xml:86]: already commented

File: appserver/payara-appserver-modules/payara-micro-cdi/pom.xml (hunk 6 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/payara-micro-cdi/pom.xml:93]: already commented

File: appserver/payara-appserver-modules/microprofile/rest-client-ssl/pom.xml (hunk 4 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/microprofile/rest-client-ssl/pom.xml:68]: already commented

File: appserver/payara-appserver-modules/microprofile/metrics/pom.xml (hunk 3 above)
Comment (breakponchito) [reply appserver/payara-appserver-modules/microprofile/metrics/pom.xml:140]: already commented

**Comment by breakponchito** [comment]
Jenkins test please

File: appserver/packager/external/jakarta-ee11-shim/pom.xml [hunk 9]
```diff
@@ -156,7 +164,45 @@
...
+                                    org.jboss.weld.injection;
+                                    org.jboss.weld.interceptor.proxy;
+                                    org.jboss.weld.interceptor.util.proxy;
+                                    org.jboss.weld.logging;
+                                    org.jboss.weld.manager;
+                                    org.jboss.weld.module.*;
+                                    org.jboss.weld.security;
+                                    org.jboss.weld.util;
+                                    org.jboss.weld.xml;
+                                    org.jboss.weld.serialization.spi;version="5.0.99";shim=true
```
Comment (Pandrex247) [inline appserver/packager/external/jakarta-ee11-shim/pom.xml:205]: I'm not convinced this works - I don't think we can have wildcards in the shim
```suggestion
                                    <!-- weld-osgi-bundle -->
                                    org.jboss.weld.bootstrap.api;
                                    org.jboss.weld.bootstrap.api.helpers;
                                    org.jboss.weld.construction.api;
                                    org.jboss.weld.context.api;
                                    org.jboss.weld.ejb.api;
                                    org.jboss.weld.manager.api;
                                    org.jboss.weld.servlet.api;
                                    org.jboss.weld.servlet.api.helpers;
                                    org.jboss.weld.bootstrap.spi;
                                    org.jboss.weld.bootstrap.spi.helpers;
                                    org.jboss.weld.configuration.spi;
                                    org.jboss.weld.configuration.spi.helpers;
                                    org.jboss.weld.ejb.spi;
                                    org.jboss.weld.ejb.spi.helpers;
                                    org.jboss.weld.injection.spi;
                                    org.jboss.weld.injection.spi.helpers;
                                    org.jboss.weld.interceptor.spi.metadata;
                                    org.jboss.weld.interceptor.spi.model;
                                    org.jboss.weld.resources.spi;
                                    org.jboss.weld.resources.spi.helpers;
                                    org.jboss.weld.security.spi;
                                    org.jboss.weld.serialization.spi;
                                    org.jboss.weld.serialization.spi.helpers;
                                    org.jboss.weld.servlet.spi;
                                    org.jboss.weld.servlet.spi.helpers;
                                    org.jboss.weld.transaction.spi;
                                    org.jboss.weld.bootstrap.event;
                                    org.jboss.weld.context;
                                    org.jboss.weld.context.activator;
                                    org.jboss.weld.context.bound;
                                    org.jboss.weld.context.ejb;
                                    org.jboss.weld.context.http;
                                    org.jboss.weld.context.unbound;
                                    org.jboss.weld.events;
                                    org.jboss.weld.inject;
                                    org.jboss.weld.interceptor;
                                    org.jboss.weld.proxy;
                                    org.jboss.weld.lite.extension.translator;
                                    org.jboss.weld.lite.extension.translator.logging;
                                    org.jboss.weld.lite.extension.translator.util;
                                    org.jboss.weld.lite.extension.translator.util.reflection;
                                    org.jboss.weld.module.ejb;
                                    org.jboss.weld.module.ejb.context;
                                    org.jboss.weld.module.ejb.context.beanstore;
                                    org.jboss.weld.module.jsf;
                                    org.jboss.weld.module.jsf.servlet;
                                    org.jboss.weld.module.jta;
                                    org.jboss.weld.module.web;
                                    org.jboss.weld.module.web.context.beanstore.http;
                                    org.jboss.weld.module.web.context.http;
                                    org.jboss.weld.module.web.el;
                                    org.jboss.weld.module.web.logging;
                                    org.jboss.weld.module.web.servlet;
                                    org.jboss.weld.module.web.util.el;
                                    org.jboss.weld.module.web.util.servlet;
                                    org.jboss.weld;
                                    org.jboss.weld.bean;
                                    org.jboss.weld.bean.builtin;
                                    org.jboss.weld.bean.proxy;
                                    org.jboss.weld.bootstrap;
                                    org.jboss.weld.config;
                                    org.jboss.weld.contexts;
                                    org.jboss.weld.contexts.activator;
                                    org.jboss.weld.contexts.bound;
                                    org.jboss.weld.contexts.cache;
                                    org.jboss.weld.contexts.conversation;
                                    org.jboss.weld.contexts.unbound;
                                    org.jboss.weld.event;
                                    org.jboss.weld.exceptions;
                                    org.jboss.weld.injection;
                                    org.jboss.weld.interceptor.proxy;
                                    org.jboss.weld.interceptor.util.proxy;
                                    org.jboss.weld.logging;
                                    org.jboss.weld.manager;
                                    org.jboss.weld.module;
                                    org.jboss.weld.security;
                                    org.jboss.weld.util;
                                    org.jboss.weld.xml;version="5.0.99";shim=true
```

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: appserver/packager/external/jakarta-ee11-shim/pom.xml (hunk 9 above)
Comment (breakponchito) [reply appserver/packager/external/jakarta-ee11-shim/pom.xml]: already added changes for the shim config, with this on my local quicklook is passing and still some issues from samples related to jersey patched version

**Review by Pandrex247 — APPROVED** [review]

**Pull request merged**